src/scripts/analytics-sw.js

```diff
@@ -21,18 +23,21 @@ class Analytics {
       return Promise.resolve();
     }
 
-    return self.registration.pushManager.getSubscription()
-    .then((subscription) => {
-      if (subscription === null) {
+    return idbKeyval.get('google-analytics-client-id')
+    .catch(() => {
+      return null;
+    })
+    .then((clientId) => {
+      if (clientId === null) {
         // The user has not subscribed yet.
-        throw new Error('No subscription currently available.');
+        throw new Error('No Client ID currently available.');
```

Comment:
In this case do you want to fail, or do you want to send the hit anonymously? Seems like it might be valuable to still have an event, even if it's not associated with the correct user.